### ISSUE TITLE
Error when uploading forms using Windows CMD/Powershell using medic-conf 3.1.0

### ISSUE THREAD

**Issue by binokaryg** [medic/medic-conf#300]
With medic-conf 3.1.0, I tried to upload contact forms using the Windows command line and Powershell. The form was uploaded, but I got this error:
```
ERROR Error: ENOENT: no such file or directory, open 'D:\MedProject\medic-projects\nepal-healthform/.hashes/form:contact:person:create.json'
    at Object.openSync (fs.js:440:3)
    at Object.writeFileSync (fs.js:1265:35)
    at Object.write (C:\Users\Binod Adhikary\AppData\Roaming\npm\node_modules\medic-conf\src\lib\sync-fs.js:101:47)
    at Object.postUploadByXml (C:\Users\Binod Adhikary\AppData\Roaming\npm\node_modules\medic-conf\src\lib\warn-upload-overwrite.js:179:6)
    at C:\Users\Binod Adhikary\AppData\Roaming\npm\node_modules\medic-conf\src\lib\upload-forms.js:63:41
    at processTicksAndRejections (internal/process/task_queues.js:94:5)
    at async module.exports (C:\Users\Binod Adhikary\AppData\Roaming\npm\node_modules\medic-conf\src\lib\main.js:188:5)
    at async C:\Users\Binod Adhikary\AppData\Roaming\npm\node_modules\medic-conf\src\bin\medic-conf.js:11:18
```
No file was created in `.hashes` directory.

Then I tried with a Bash shell (Git Bash, WSL bash), and the command ran successfully without errors.

The files created in `.hashes` directory after running the command successfully from bash in Windows had a different symbol instead of a colon (expected `form:contact:person:create.json`):

![image](https://user-images.githubusercontent.com/20041732/76943295-36aa6880-6927-11ea-8cc8-9813d27c2407.png)

I guess this error occurs because Windows [does not support](https://docs.microsoft.com/en-us/windows/win32/fileio/naming-a-file) colons in the filenames.

**Comment by kennsippell** [comment]
This bug causes a crash when running the command `medic-conf --local` on windows machines. Has this been prioritised correctly?

**Comment by kennsippell** [comment]
https://forum.communityhealthtoolkit.org/t/core-framework-easy-deployment-instructions-error/590/7

**Comment by garethbowen** [comment]
> Has this been prioritised correctly?

I believe so, yes.

> What is our recommended workaround for windows users?

Deleting the `.hashes` directory should bypass the hash checking until this is fixed.

**Comment by binokaryg** [comment]
Deleting the `.hashes` directory does not help when uploading contact forms from Windows command line or PowerShell. The error still shows up after uploading a form.

If we are uploading multiple forms at once, only the first form gets uploaded, then it crashes.
```
WARN Error reading file: D:\work\medic\ssh-git\medic-projects\nepal-healthform/.hashes/form:contact:clinic:create.json
INFO Uploaded form D:\work\medic\ssh-git\medic-projects\nepal-healthform/forms/contact/clinic-create.xml
ERROR Error: ENOENT: no such file or directory, open 'D:\work\medic\ssh-git\medic-projects\nepal-healthform/.hashes/form:contact:clinic:create.json'
    at Object.openSync (fs.js:443:3)
    at Object.writeFileSync (fs.js:1194:35)
    at Object.write (C:\Users\binok\AppData\Roaming\npm\node_modules\medic-conf\src\lib\sync-fs.js:101:47)
    at Object.postUploadByXml (C:\Users\binok\AppData\Roaming\npm\node_modules\medic-conf\src\lib\warn-upload-overwrite.js:179:6)
    at promiseChain.then.then.then.then (C:\Users\binok\AppData\Roaming\npm\node_modules\medic-conf\src\lib\upload-forms.js:63:41)
    at process._tickCallback (internal/process/next_tick.js:68:7)
```

**Comment by garethbowen** [comment]
I don't have a Windows machine to test this out on, but some other possible workarounds are...

1. Use the Windows Subsystem for Linux to run medic-conf in a Linux environment. This may not work due to the underlying filesystem being Windows, though I understand WSL2 is even more Linux-like, so has a better chance of working.
2. Use an older version of medic-conf but this may cause issues with uploading incorrect config if you're uploading to a recent version of cht-core.

**Comment by kennsippell** [comment]
> Use an older version of medic-conf but this may cause issues with uploading incorrect config if you're uploading to a recent version of cht-core.

medic-conf 3.1 is required for 3.8.0 instances. This is not a valid workaround for a growing cohort of users.

Can we revert this feature until it is stable? Or put it behind a command-line flag like `--safe-upload` so people can opt in to the experience?

**Comment by garethbowen** [comment]
This is the feature: https://github.com/medic/medic-conf/issues/113

If the configuration process is being followed correctly then it shouldn't block anyone. If the configuration process is _not_ being followed then it'll warn but ultimately you can choose to overwrite other changes.

Other than this Windows specific issue, how is this disrupting work? There are bound to be other improvements we can make.

Adding a `--safe-upload` flag would be akin to removing the feature altogether, which is an option if it provides no actual value.

**Comment by garethbowen** [comment]
Ready for AT in `301-check-properties-too`

This branch fixes: #271 #300 #301 #303 #306 so AT them at the same time.

**Comment by ngaruko** [comment]
LGTM .
Also double-checked with @binokaryg !
